### DISCUSSION TITLE
use multi-domain-cert in traefik (with scheduler-k3s)

### DISCUSSION THREAD

**Discussion by dirkzado** [dokku/dokku#6962] [Q&A]
I am trying to figure out how to use our multi-domain-cert within traefik (which is the default for scheduler-k3s).
Actually my issue seems to be to use a custom cert together with traefik within scheduler-k3s (no letsencrypt)?

I added the cert with:
`dokku certs:add quickshift-preview /path/to/cert/fullchain.pem /path/to/cert/privkey.pem`
(and the cert is in the path and readable)

This works fine as long as I use nginx (https://dokku.com/docs/configuration/ssl/)

Since I like to use multiple servers with scheduler-k3s. I had to use traefik. But I can not figure how use a custom (multi-domain) cert with dokku and traefik.  Is there a solution?

I just found: "The traefik plugin only supports automatic ssl certificates from it's letsencrypt integration." (https://dokku.com/docs/networking/proxies/traefik/)

Is there a possibility to use a traefik config like it is described here: https://doc.traefik.io/traefik/https/tls/

Any hint is welcome! ;-) Thank you!

**Comment by josegonzalez** [comment]
Currently the scheduler-k3s plugin does not support integrating with the certs plugin (either for traefik or nginx). It's not super clear from the traefik docs how we could reference a certificate - I'd assume we could store it as a secret and somehow annotate the service or something, but again, not clear.

Is this something you can research and get back to us on? Then I can take that and implement it :)

**Comment by dirkzado** [comment]
Thank you for you answer!

I found a different approach for us: we're now using Dockerfile to build docker images and put it to a private docker hub repository.
From there every app server can get it through `dokku git:from-image <dokku-app> dockerhub/dokku-app:latest` and I still can use nginx which supports multi-cert-domain quite good ;)
